libs/config/src/analyze/lints/c12_math_could_be_unquoted.rs

```diff
@@ -0,0 +1,174 @@
+use std::{ops::Range, sync::Arc};
+
+use hemtt_common::config::{LintConfig, LintEnabled, ProjectConfig};
+use hemtt_workspace::{
+    lint::{AnyLintRunner, Lint, LintRunner},
+    reporting::{Code, Diagnostic, Processed, Severity},
+};
+
+use crate::{analyze::LintData, Item, Number, Value};
+
+crate::analyze::lint!(LintC12MathCouldBeUnquoted);
+
+impl Lint<LintData> for LintC12MathCouldBeUnquoted {
+    fn ident(&self) -> &'static str {
+        "math_could_be_unquoted"
+    }
+
+    fn sort(&self) -> u32 {
+        120
+    }
+
+    fn description(&self) -> &'static str {
+        "Reports on quoted math statements that could be evaulated at build-time"
+    }
+
+    fn documentation(&self) -> &'static str {
+        "### Example
+
+**Incorrect**
+```hpp
+x = '1+1';
+```
+
+**Correct**
+```hpp
+x = 1+1; // hemtt will evaluate at build-time to 2
+```
+
+### Explanation
+Quoted math statements will have to be evaulated in-game
+"
+    }
+
+    fn default_config(&self) -> LintConfig {
+        // false-positives are possible
+        LintConfig::help().with_enabled(LintEnabled::Pedantic)
```

Comment:
You've seen false positives from this? I would think it should be possible to avoid them

Reply:
just possible in theory as some kind of formatted text
like `text = "0000-9999" // input range`

Reply:
Ah, true

Reply:
Personally I think this would be fine to enable by default, I think the odds of encountering text like that is low enough that the benefit is worth it